### PR TITLE
fix consumer subscribe bug

### PR DESCRIPTION
**Pull request by rkatsyuryna**

### CODE REVIEW

File: core/src/main/java/io/github/tcdl/Consumer.java [hunk 1]
```diff
@@ -60,7 +59,12 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
      */
     public void subscribe(MessageHandler messageHandler) {
         Validate.notNull(messageHandler, "the 'messageHandler' must not be null");
-        this.messageHandler = messageHandler;
+        if(this.messageHandler == null) {
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Consumer.java]: missing whitespace

File: core/src/main/java/io/github/tcdl/Consumer.java [hunk 2]
```diff
@@ -60,7 +59,12 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
      */
     public void subscribe(MessageHandler messageHandler) {
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Consumer.java]: should it be synchronized?

File: core/src/main/java/io/github/tcdl/Consumer.java [hunk 3]
```diff
@@ -60,7 +59,12 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
...
         Validate.notNull(messageHandler, "the 'messageHandler' must not be null");
-        this.messageHandler = messageHandler;
+        if(this.messageHandler == null) {
+            this.messageHandler = messageHandler;
+            this.rawAdapter.subscribe(this::handleRawMessage);
+        } else {
+          throw new IllegalStateException("Message handler already set for topic " + topic);
+        }
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Consumer.java]: Should we set messageHandler to null in end() as well?

File: core/src/main/java/io/github/tcdl/Consumer.java (hunk 2 above)
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/Consumer.java]: find other way roung

**Pull request closed without merging**